tools/CodeFenceChecker/Collector.fs

```diff
@@ -0,0 +1,65 @@
+//
+// Permission to use, copy, modify, and/or distribute this software for any purpose
+// with or without fee is hereby granted.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+// REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+// INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+// LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE
+// OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+// PERFORMANCE OF THIS SOFTWARE.
+//
+namespace CodeFenceChecker
+
+[<AutoOpen>]
+module Collector =
+    open System
+    open System.IO
+    open System.Text.RegularExpressions
+
+    let nl = Environment.NewLine
+    let sep = Path.DirectorySeparatorChar
+    let isEmptyLine (line: string) = String.IsNullOrEmpty(line.Trim())
+
+    let private matchBlockStart line = Regex.IsMatch(line, @"^.*[`~]{3,}(fsharp)")
+    let private matchBlockEnd line = Regex.IsMatch(line, @"^.*[`~]{3,}$")
```

Comment:
At some in the future, if you'd be interested, you could consider using parsing the Markdown using a library like [Markdig](https://github.com/xoofx/markdig) to find the code fence blocks (one can use an arbitrary number of backticks).